drivers/dma/dma_silabs_siwx91x_gpdma.c

```diff
@@ -428,32 +453,57 @@ bool siwx91x_gpdma_chan_filter(const struct device *dev, int channel, void *filt
 	}
 }
 
-static int siwx91x_gpdma_init(const struct device *dev)
+static int gpdma_siwx91x_pm_action(const struct device *dev, enum pm_device_action action)
 {
 	const struct siwx91x_gpdma_config *cfg = dev->config;
 	struct siwx19x_gpdma_data *data = dev->data;
-	RSI_GPDMA_INIT_T gpdma_init = {
-		.pUserData = NULL,
-		.baseG = (uint32_t)cfg->reg,
-		.baseC = (uint32_t)cfg->channel_reg,
-		.sramBase = (uint32_t)data->desc_pool->buffer,
-	};
-	RSI_GPDMA_HANDLE_T gpdma_handle;
 	int ret;
 
-	ret = clock_control_on(cfg->clock_dev, cfg->clock_subsys);
-	if (ret) {
-		return ret;
-	}
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		break;
+	case PM_DEVICE_ACTION_SUSPEND:
+		break;
+	case PM_DEVICE_ACTION_TURN_ON:
+		RSI_GPDMA_INIT_T gpdma_init = {
+			.pUserData = NULL,
+			.baseG = (uint32_t)cfg->reg,
+			.baseC = (uint32_t)cfg->channel_reg,
+			.sramBase = (uint32_t)data->desc_pool->buffer,
+		};
+		RSI_GPDMA_HANDLE_T gpdma_handle;
```

Comment:
A `case` does not open a new scope. So these declarations are not on top of any scope.